new-docs/content/any/getting-started/policies/index.md

```diff
@@ -9,325 +9,401 @@ description: Learn about writing Oso policies - the source of truth for authoriz
-{{< /code >}}
-
-<!-- TODO(gj): Link `Oso.isAllowed()` once API docs are setup. -->
-
-The above rule expects the `actor` variable to be a {{% exampleGet "langName"
-%}} {{% exampleGet "instance" %}} with the field `{{% exampleGet "isAdmin" %}}`. The {{% exampleGet "langName" %}} {{% exampleGet "instance" %}} is passed
-into Oso with a call to `Oso.{{% exampleGet "isAllowed" %}}()`:
-
-{{% exampleGet "userClass" %}}
-
-The code above provides a `User` object as the _actor_ for our `allow` rule.
-Since `User` has a field called `{{% exampleGet "isAdmin" %}}`, it is checked
-during evaluation of the Polar rule and found to be true.
-
-In addition to accessing attributes, you can also call methods on application
-instances in a policy:
-
-{{< code file="policy.polar" >}}
-allow(actor, action, resource) if actor.{{% exampleGet "isAdminOf" %}}(resource);
-{{< /code >}}
-
-### Registering Application Types
-
-Instances of application types can be constructed from inside an Oso policy
-using [the `new` operator](polar-syntax#new) if the class has been
-**registered**. {{% exampleGet "registerClass" %}}
-
-Once the class is registered, we can make a `User` object in Polar. This can be
-helpful for writing inline test queries:
-
-{{% exampleGet "testQueries" %}}
-
-Registering classes also makes it possible to use
-[specialization](polar-syntax#specialization) and [the `matches`
-operator](polar-syntax#matches-operator) with the registered class.
-
-In our previous example, the **allow** rule expected the actor to be a `User`,
-but we couldn’t actually check that type assumption in the policy. If we
-register the `User` class, we can write the following rule:
-
-{{< code file="policy.polar" >}}
-allow(actor: User, action, resource) if actor.name = "alice";
-{{< /code >}}
-
-This rule will only be evaluated when the actor is a `User`; the `actor`
-argument is _specialized_ on that type. We could also use `matches` to express
-the same logic on an unspecialized rule:
+```polar
+allow(actor: User, "GET", resource: Expense) if ...;
+```
 
-{{< code file="policy.polar" >}}
-allow(actor, action, resource) if actor matches User{name: "alice"};
-{{< /code >}}
+### Built-in Classes
 
-Either way, using the rule could look like this:
+We said above that any class you wish to use as a specializer
+(or on the right-hand side of the `matches` operator) must be
+registered with Polar. For convenience, Polar automatically
+registers a few classes, such as:
 
-{{% exampleGet "specializedExample" %}}
+* `Dictionary`
+* `String`
+* `List`
+* `Integer`
+* `Float`
 
-{{% callout "Note" "green" %}}
-Type specializers automatically respect the **inheritance** hierarchy of
-application classes. See the [Resources with
-Inheritance](guides/more/inheritance) guide for an in-depth
-example of how this works.
-{{% /callout %}}
+These classes correspond to ones in the application language,
+e.g., in Python, `String` is actually the built-in `str` class.
```

Comment:
We can link to appropriate reference docs here.

Reply:
Still think this would be good, but again it requires language-specific docs for each, so would do this last if time.